package/MDAnalysis/core/bondtable.pyx

```diff
@@ -0,0 +1,760 @@
+
+        Parameters
+        ----------
+        target: int
+            The index of the atom to get bonds for.
+        
+        Returns
+        -------
+        pairs: list
+            A list of bonds that contain the target index 
+        """
+        cdef vector[cpair[int, int]] pairs
+        cdef cpair[vector[cpair[int, int]], cbool] ret_struct
+        if self._is_empty:
+            return []
+        if target <= self.max_index:
+            ret_struct = self._get_pair(target)
+            if ret_struct.second:
+                pairs = ret_struct.first
+                return pairs
+            else:
+                return []
+        else:
+            return []
+
+    def get_bonds_slice(self, targets):
+        """
+        Get bonded atoms for an array of indices.
+
+        Parameters
+        ----------
+        targets: np.ndarray or scalar index
+            the atoms to get bonded atoms for 
+
+        Returns
+        -------
+        bonds: list
+            The bonded atoms for the indices in targets
+        """
+        if self._is_empty:
+            return np.empty((0), dtype=np.int32)
+        if np.isscalar(targets):
+            targets = np.asarray([targets])
+        cdef vector[vector[int]] bonds
+        cdef vector[int] row, tmp
+        cdef int i, j, size_j
+        for i in range(targets.shape[0]):
+            index = targets[i]
+            if index <= self.max_index:
+                row = self._get_bond(index)
+                for j in range(row.size()):
+                    size_j += 1
+                    tmp = vector[int](2)
+                    tmp[0] = targets[i]
+                    tmp[1] = row[j]
+                    bonds.push_back(tmp)
+            else:
+                pass
+        return np.asarray(bonds, dtype=np.int32)
+
+    def get_pairs_slice(self,  targets):
+        """
+        Get the bond pairs for an array of indices
+        
+        Parameters
+        ----------
+        targets: np.ndarray or scalar index
+            the atoms to get bond pairs for
+
+        Returns
+        -------
+        pairs: np.ndarray
+            An array of bonds that contain the target index 
+        """
+        if self._is_empty:
+            return np.empty((0), dtype=np.int32)
+        if np.isscalar(targets):
+            targets = np.asarray([targets])
+        cdef vector[cpair[int, int]] bonds
+        cdef vector[cpair[int, int]] pair_arr
+        cdef cpair[vector[cpair[int, int]], cbool] ret_struct
+        cdef cbool has_bonds
+        cdef int i, j
+        for i in range(targets.shape[0]):
+            if targets[i] <= self.max_index:
+                ret_struct = self._get_pair(targets[i])
+                pair_arr = ret_struct.first
+                has_bonds = ret_struct.second
+                if has_bonds:
+                    for j in range(pair_arr.size()):
+                        bonds.push_back(pair_arr[j])
+            else:
+                pass
```

Comment:
There's a bunch of explicit `else: pass` blocks--are those placeholders or just a stylistic preference over not writing the `else` at all?

Reply:
Stylistic but I can change it if people don't like it, I know it looks a bit strange.